File: Blind 75/1. Array Problems/1.two_sum.py
#https://leetcode.com/problems/two-sum/

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map={}
        for i in range(len(nums)):
            d= target-nums[i]
            if d in map:
                return [map[d],i]
            else:
                map[nums[i]]=i
                    

File: Blind 75/1. Array Problems/1. TwoSum.java
//https://leetcode.com/problems/two-sum/

import java.util.Map;
import java.util.HashMap;

class Solution {
    
    Map<Integer,Integer> map = new HashMap<>();
    
    public int[] twoSum(int a[], int x) {
                
        for(int i=0;i<a.length;i++){
            
            if(!map.containsKey(x-a[i]))
            	map.put(a[i],i);
            else
            	return new int[] {map.get(x-a[i]),i};
        }
    	return new int[] {};
    }
}